src/Formats/NativeReader.cpp

```diff
@@ -145,12 +146,17 @@ Block NativeReader::read()
         readBinary(type_name, istr);
         column.type = data_type_factory.get(type_name);
 
-        const auto * aggregate_function_data_type = typeid_cast<const DataTypeAggregateFunction *>(column.type.get());
-        if (aggregate_function_data_type && aggregate_function_data_type->isVersioned())
+        auto callback = [&](DataTypePtr & type)
         {
-            auto version = aggregate_function_data_type->getVersionFromRevision(server_revision);
-            aggregate_function_data_type->setVersion(version, /*if_empty=*/ true);
-        }
+            const auto * aggregate_function_data_type = typeid_cast<const DataTypeAggregateFunction *>(type.get());
+            if (aggregate_function_data_type && aggregate_function_data_type->isVersioned())
+            {
+                auto version = aggregate_function_data_type->getVersionFromRevision(server_revision);
+                aggregate_function_data_type->setVersion(version, /*if_empty=*/ true);
```

Comment:
Seems like it should be done by just one method in `DataTypeAggregateFunction`. I'd name it like `updateVersionFromRevision`.